JsonLedgerApiSample/src/test/java/com/example/signing/TopologyTransactionHashBuilderTest.java

```diff
@@ -0,0 +1,43 @@
+package com.example.signing;
+
+import com.daml.ledger.api.v2.interactive.InteractiveSubmissionServiceOuterClass;
+import com.example.services.Ledger;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+class TopologyTransactionHashBuilderTest {
+
+    void testFromBase64Payload(String expectedOutputBase64, List<String> inputs) throws Exception {
+        byte[] expectedOutputRaw = Encode.fromBase64String(expectedOutputBase64);
+        byte[] actualOutputRaw = new TopologyHashBuilder(inputs).hash();
+
+        if (!Arrays.equals(expectedOutputRaw, actualOutputRaw)) {
+            String base64ComputedHash = Encode.toBase64String(actualOutputRaw);
+            throw new IllegalStateException("Topology hash mismatch: %s (provided) vs %s (computed) for topology tranasctions [%s]"
+                    .formatted(expectedOutputBase64, base64ComputedHash, String.join(", ", inputs)));
+        }
+    }
+
+    @Test
+    void createTreasuryExample() throws Exception {
+        List<String> inputs = new ArrayList<>();
+        inputs.add("CosBCAEQARqEAQqBAQpEMTIyMDllYzljMTQxY2M3MGZiNjRiOTg3M2EyZWY3ZGQ3NmY5YjUxYzFhOWE5MjA3ZmJmMmI3NjhlMDQwODdkZDQyNWISNxAEGiwwKjAFBgMrZXADIQBhJWW3y52wD+i3fB7JVAb10zZK+ViNw4SbjT8G8MdYWyoDAQUEMAEiABAe");
```

Comment:
Am I right, that here we must use the base64 encoded strings? no chance for readable JSON?

Independently, add comments what they represent and where the test cases came from.